src/assets/Keycloak/Keycloak.md

```diff
@@ -0,0 +1,147 @@
+Additionally, it enables users to authenticate through various identity providers and use fine-grained permissions for regulating access to Software as a Service (SaaS) applications.
+
+It facilitates the creation of a user database with customizable roles and groups, offering functionalities such as user management, registration, and password policy enforcement.
+This makes it a comprehensive Identity and Access Management solution for contemporary applications. Keycloak API enables the integration of all these features into your application
+without the need for additional coding.
+
+## Keycloak Features
+![Keycloak features](/src/assets/Keycloak/images/Keycloak/Keycloak-features.png)
+
+1. Identity Broker:
+   Identity Broker acts as a middleman, linking various service providers with different identity providers. It establishes trust with external identity providers to utilize their identities for accessing internal services offered by service providers.
+   Imagine a textile company, which has two Keycloak setups: one for customer and another for internal employees. If you want your employees to access external services without creating new accounts on customer Keycloak, you can use the customer Keycloak
+   as `Broker` and the internal Keycloak as `Provider` in this scenario.
+
+2. User Federation:
+   Keycloak provides the ability to integrate with existing LDAP or Active Directory servers. It also supports implementation of your own provider such as a relational database.
+
+3. Standard Protocols:
+   Keycloak is based on three standard protocols i.e., OpenID Connect, OAuth2.0, and SAML.
+
+4. Password Policies:
+   Keycloak contains different password policies i.e., HashAlgorithm, Hashing Iterations, Digits, Lowercase Characters, Regular Expression and so on.
+
+5. Single Sign-On:
+   Keycloak facilitates a seamless login experience where users only need to sign in once to access multiple applications, eliminating the need for repeated logins.
+   This not only streamlines user experience but also simplifies identity management for administrators.
+
+6. Social Login:
+   Keycloak has built-in support to login via Google, GitHub, and Facebook, which helps us to use social identity providers.
+
+## Keycloak Drawbacks
+Despite its extensive features, Keycloak does have certain limitations. One of these is the need for a more varied implementation approach.
+Additionally, the following are some of the drawbacks associated with Keycloak:
+1. Complex Server Deployment: The manual process involved in deploying Keycloak on a server can be complicated, potentially affecting overall productivity.
+2. Lack of Comprehensive Support Documentation: Despite its robust functionalities, Keycloak could benefit from more efficient and extensive support documentation. This would make it easier for users to find suitable solutions to their problems.
+
```

Comment:
Usually the negative points are mentioned towards the end of a blog post. It makes sense though to mention them here before going into the details of installation. However for the flow of the text it might help to add one sentence to transition to the next paragraph. Something along the lines that it is not the perfect solution for every situation, but still recommended for many scenarios and encouraging readers to learn more about how to install and use it for their projects...

Reply:
Upto my knowledge step depends upon services integrating with Keycloak. oCIS, Github, Nextcloud has separate step.

Reply:
I think the order will be good enough for this blog. Next time while I target academic people, I surely remember this to make my transition fluent.